.github/workflows/ruby.yml

```
@@ -0,0 +1,37 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['2.7']
+
+    steps:
+    - uses: actions/checkout@v2
```

```suggestion
    - uses: actions/checkout@v2.3.4
```

This is the latest stable release, see the release notes here:
https://github.com/actions/checkout/releases/tag/v2.3.4

---

Or we could get rid of all versions and always use the latest one.

---

https://github.com/actions/checkout/tags shows that `v2` and `v.2.3.4` point to the same commit. I cannot see any automation on the `action/checkout` repository with regards to the tags. I think updating the tags at release is a manual chore, that can be forgotten by the maintainers. That means you depend on the `actions/checkout` project to update the `v2` tag when they make a new release. If they forget to update the tag, you're using an outdated version, without knowing about it.

When you use Dependabot to update the `github-action` runners, you always get the update in a reasonable time-frame (at most 3 days, in case of release of new runner on Saturday, then Dependabot will next run on Mondays).
Another benefit of using Dependabot is that it shows you the change-log for the github actions runners. This way you're aware of what the project is doing.

---

That's adding a lot of commits just to keep track of external changes. I strongly prefer to rely on latest tagged version, with correct semantic versioning.

---

My understanding of Actions best practices is that we should just use `v2` unless we have a specific need for a more exact version, and then we'll pick up bug fixes or other improvements along the way. It may be that some Actions do not follow semantic versioning correctly, but I would expect the official `actions/checkout` one to do so.